### PR TITLE
[ISSUE #3346] add  muti values 

### LINKED ISSUE #3346: [Task] refactor the places which maybe need muti values
### Description

Now `ParameterData#builder` has only one return value, which is not rigorous enough in some cases (such as `cookie`and `header`). You need to find these places (maybe not only `ParameterData#builder` has this problem) and refactor it. If you want to contribute or find places similar to `ParameterData#builder`, please leave a message and I will assign it to you.

### Task List

1.  - [ ] refactor `org.apache.shenyu.plugin.base.condition.data.ParameterData#builder`;
2. - [ ] refactor `org.apache.shenyu.plugin.base.condition.data.HeaderParameterData`;
3. - [ ] refactor `org.apache.shenyu.plugin.base.condition.data.CookieParameterData`;
4. - [ ] find places which maybe need muti values;

### PR DESCRIPTION
**Pull request by ChineseTony**
// Describe your PR here; eg. Fixes #3346 

- [ ] You have read the [contribution guidelines](https://shenyu.apache.org/community/contributor-guide).
- [ ] You submit test cases (unit or integration tests) that back your changes.
- [ ] Your local test passed `mvn clean install -Dmaven.javadoc.skip=true`.

### CODE REVIEW

File: shenyu-plugin/shenyu-plugin-base/src/main/java/org/apache/shenyu/plugin/base/condition/data/PostParameterData.java [hunk 1]
```diff
@@ -17,21 +17,30 @@
...
     
     @Override
-    public String builder(final String paramName, final ServerWebExchange exchange) {
+    public List<String> builder(final String paramName, final ServerWebExchange exchange) {
         ShenyuContext shenyuContext = exchange.getAttribute(Constants.CONTEXT);
-        return (String) ReflectUtils.getFieldValue(shenyuContext, paramName);
+        List<String> result = Lists.newArrayList();
+        String parameter = (String) ReflectUtils.getFieldValue(shenyuContext, paramName);
+        if (StringUtils.isNotEmpty(parameter)) {
+            result.add((String) ReflectUtils.getFieldValue(shenyuContext, paramName));
```
Comment (dragon-zhang) [inline shenyu-plugin/shenyu-plugin-base/src/main/java/org/apache/shenyu/plugin/base/condition/data/PostParameterData.java:42]: why not use `parameter` directly?

File: shenyu-plugin/shenyu-plugin-base/src/main/java/org/apache/shenyu/plugin/base/condition/judge/PredicateJudgeFactory.java [hunk 2]
```diff
@@ -56,6 +58,30 @@ public static Boolean judge(final ConditionData conditionData, final String real
...
+        if (Objects.isNull(conditionData)) {
+            return false;
+        }
+        if (CollectionUtils.isEmpty(realDatas)) {
+            return false;
+        }
+        PredicateJudge predicateJudge = newInstance(conditionData.getOperator());
+        Boolean flag = true;
+        for (String realData:realDatas) {
+            flag = predicateJudge.judge(conditionData, realData);
```
Comment (dragon-zhang) [inline shenyu-plugin/shenyu-plugin-base/src/main/java/org/apache/shenyu/plugin/base/condition/judge/PredicateJudgeFactory.java:77]: should also consider `StringUtils.isBlank(realData)`.

File: shenyu-plugin/shenyu-plugin-base/src/test/java/org/apache/shenyu/plugin/base/condition/judge/PredicateJudgeFactoryTest.java [hunk 3]
```diff
@@ -51,7 +59,7 @@ public void testConditionDataIsNull() {
 
     @Test
     public void testRealDataIsBlank() {
-        assertFalse(PredicateJudgeFactory.judge(conditionData, null));
```
Comment (dragon-zhang) [inline shenyu-plugin/shenyu-plugin-base/src/test/java/org/apache/shenyu/plugin/base/condition/judge/PredicateJudgeFactoryTest.java:54]: I think this line can be kept. Check whether there are other `null test` lines that can be kept.

**Review by yu199195 — CHANGES_REQUESTED** [review]
not merged

**Comment by yu199195** [comment]
@ChineseTony thanks for you contributor， There is a bit of a problem with the modifications to this feature, thank you again！

**Pull request closed without merging**